modules/accredited_representative_portal/app/models/accredited_representative_portal/saved_claim/benefits_intake.rb

```diff
@@ -59,6 +59,18 @@ module BusinessLines
       end
 
       delegate :to_pdf, to: :form_attachment
+
+      def parsed_form
+        JSON.parse(form)
+      end
+
+      def claimant_info
+        parsed_form.key?('dependent') ? parsed_form['dependent'] : parsed_form['veteran']
+      end
```

Comment:
Thoughts on strongly eschewing logic away from the model layer?

I tend to find that both Rails models and controllers can become overburdened or overly coupled to and that this is what motivates the missing Rails service layer. But if we don't feel like making a service layer here to perform business logic transformations derived from request params above it and database state below it, moving to the serializer can help prevent us from future accidental coupling.

In particular we have a way to combine 2x properties of the root object we're serializing to achieve this transformation there. Something like:
```ruby
object.saved_claim.parsed_form[object.claimant_type]
```

Reply:
Interestingly, having `claimant_type` separate from the record that has the corresponding data has two slightly different reasons for claim submissions and POA requests. For POA requests it is because we completely delete the form record as our implementation of redaction, so when that property is cached on the parent we can retain that information post-redaction. In this case it seems to be a bit more driven by a desire to not need to modify the original table `saved_claims` that lots of people use.

I do think it is a little funky... But I think that is some of the reasoning.